core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -138,7 +138,7 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
               if (shouldShowCompletedJobs) {
                 <li>
                   <a href="#completed"><strong>Completed Jobs:</strong></a>
-                  {completedJobs.size}
+                  {listener.numCompletedJobs}
```

Comment:
When the number of completed jobs is greater than the retention limits, this will now show the total number of jobs ever run rather than capping out at 1000.  I think I remember someone suggesting that we say something like "(Completed jobs: 3000 (only showing 1000))" in these cases, but even without doing that I think this is fine since I think it's less confusing than the current behavior.